### PR TITLE
Remove list_size struct member from list implementation

### PR DESCRIPTION
**Pull request by MigeljanImeri**
### Motivation and Context

In a previous pull request I opened (#15478), I added an additional counter to track the number of pending sync IO operations as it wasn't properly being tracked before. This adds extra space requirements to the vdev_queue struct and to counteract this increase it was suggested to remove the list_size struct member as it was only used in a few asserts in the code. 
### Description

I removed the list_size struct member in the list implementations in two separate commits, in order to keep the kernel list and the user-space list changes separate, in case this has to be reverted in the future. 
### How Has This Been Tested?

I ran zloop.sh and no apparent crashes were seen. 

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Library ABI change (libzfs, libzfs\_core, libnvpair, libuutil and libzfsbootenv)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the OpenZFS [code style requirements](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/openzfs/zfs/tree/master/tests) to cover my changes.
- [x] I have run the ZFS Test Suite with this change applied.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/openzfs/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

**Review by bwatkinson — CHANGES_REQUESTED** [review]
Make sure to fix the checksytle error.

I do wonder if it is worth going into each call sight for `list_create()` and removing the passed second size parameter? That might be overkill though. I will leave it up to others to decide if that should also be done.

File: lib/libspl/list.c [hunk 1]
```diff
@@ -62,10 +62,9 @@ void
 list_create(list_t *list, size_t size, size_t offset)
 {
 	ASSERT(list);
-	ASSERT(size > 0);
-	ASSERT(size >= offset + sizeof (list_node_t));
```
Comment (amotin) [inline lib/libspl/list.c:66]: I would keep these and below assertions to keep future code compatible with Illumos or whatever other port that may still use the size until we decide to remove it completely.

**Review by amotin — APPROVED** [review]
Thanks. Just one comment, but I do not insist.

File: lib/libspl/list.c (hunk 1 above)
Comment (MigeljanImeri) [reply lib/libspl/list.c]: Okay, I've added those assertions back in.

**Review by bwatkinson — APPROVED** [review]

**Review by amotin — APPROVED** [review]

**Review by behlendorf — APPROVED** [review]

**Comment by behlendorf** [comment]
When merging I intentionally squashed these two commits.  This is such a small change I don't think we need to leave the kernel and user portions split just in case we end up wanting to revert part of it.

**Pull request merged**